### PR TITLE
Add support for DoT to DNS probes

### PR DESCRIPTION
**Pull request by edevil**
This adds a new DNS probe option `dns_over_tls` to enable DoT on DNS probes.

### CODE REVIEW

**Comment by brian-brazil** [comment]
Hmm, I can see this being confusing if it silently doesn't do anything when UDP is asked for.

Docs also need updating.

**Comment by edevil** [comment]
@brian-brazil do you want me to error out when UDP and DoT is chosen? I’ll update the docs shortly.

**Comment by brian-brazil** [comment]
An error would do it.

**Comment by edevil** [comment]
@brian-brazil Added error case, and docs.

File: CONFIGURATION.md [hunk 1]
```diff
@@ -153,6 +153,9 @@ tls_config:
 
 [ transport_protocol: <string> | default = "udp" ] # udp, tcp
 
+# Whether to use DNS over TLS. This only makes sense with TCP transport protocol.
```
Comment (brian-brazil) [inline CONFIGURATION.md:156]: This only works with TCP.

Comment (edevil) [reply CONFIGURATION.md]: Done.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**